src/main/java/vilksp/returnedService/model/PaymentCase.java

```diff
@@ -19,30 +16,40 @@ public class PaymentCase {
     private Payment payment;
 
     private LocalDateTime createdAt;
+    private boolean solved;
 
-    public PaymentCase(Long l, Payment eur) {
+    private PaymentCase(Payment payment) {
+        this.payment = payment;
+        this.createdAt = LocalDateTime.now();
+        this.solved = false;
     }
 
     public PaymentCase() {
     }
 
-    public LocalDateTime getCreatedAt() {
-        return createdAt;
+    public static PaymentCase create(Payment payment) {
+        return new PaymentCase(payment);
     }
 
-    public void setCreatedAt(LocalDateTime createdAt) {
-        this.createdAt = createdAt;
+    public void solve() {
```

Comment:
Renaming a method from `setSolved` to `solve` and moving one rule inside it is a good start, but it doesn't make the domain model rich. To make it rich, most (perhaps all) of the business logic should be moved from services into entities.